doc/reference/reference_rock/vshard/vshard_storage.rst

```diff
@@ -310,6 +311,59 @@ Storage public API
             ck_constraint: []
         ...
 
+..  _storage_api-on_bucket_event:
+
+..  function:: vshard.storage.on_bucket_event([trigger-function[, old-trigger-function]])
+
+    Since vshard v.0.1.22. Define a trigger for execution when the data from
+    the user spaces is changed (deleted or inserted) due to the rebalancing
+    process. It's invoked on every data batch changed.
+
+    :param trigger-function: function which will become the trigger function.
+    :param old-trigger-function: existing trigger function which will
+                                 be replaced by trigger-function.
+
+    :return: nil or function pointer
+
+    The trigger-function can have up to three parameters:
+
+      * (string) event_type. In order to distinguish event you could
+        compare this argument with supported event types: ``bucket_data_recv_txn``
+        and ``bucket_data_gc_txn``;
+      * (unsigned) bucket_id;
+      * (table) data. E.g. includes an array of all spaces (data.spaces),
+        affected by a transaction, in which trigger-function is executed.
+
+    **Example:**
+
+    ..  code-block:: lua
+
+        vshard.storage.on_bucket_event(function(event, bucket_id, data)
+            if event == 'bucket_data_recv_txn' then
+                -- Handle it.
+                for idx, space in ipairs(data.spaces) do
+                    ...
+                end
+            elseif event == 'bucket_data_gc_txn' then
+                -- Handle it.
+                ...
+            end
+        end)
+
+    .. NOTE::
+
+        As everything executed inside triggers is already in a transaction,
+        you shouldn't use transactions, yield-operations, changes to different
+        space engines.
+
+        If the parameters are ``(nil, old-trigger-function)``,
+        then the old trigger is deleted.
+
+        If both parameters are omitted, then the response is a list of
+        existing trigger functions.
```

Comment:
```suggestion
    ..  NOTE::

        As everything executed inside triggers is already in a transaction,
        you shouldn't use transactions, yield-operations
        (:ref:`explicit <app-yields>` or not), or changes 
        to different space engines (see :ref:`rule #2 <box-txn_management>`).

    If the parameters are ``(nil, old-trigger-function)``,
    then the old trigger is deleted.

    If both parameters are omitted, then the response is a list of
    existing trigger functions.
```

Reply:
This one is pretty default note for triggers, e.g.: https://www.tarantool.io/en/doc/latest/reference/reference_lua/net_box/#net-box-on-schema-reload.

I'd rather let all info remain inside note, if you don't mind. If you insist, I'll apply.